app/src/main/java/com/duckduckgo/app/di/VariantModule.kt

```diff
@@ -33,11 +34,11 @@ object VariantModule {
     @SingleInstanceIn(AppScope::class)
     fun variantManager(
         statisticsDataStore: StatisticsDataStore,
-        weightedRandomizer: WeightedRandomizer,
+        weightedRandomizer: IndexRandomizer,
         appBuildConfig: AppBuildConfig,
     ): VariantManager =
         ExperimentationVariantManager(statisticsDataStore, weightedRandomizer, appBuildConfig)
 
     @Provides
-    fun weightedRandomizer() = WeightedRandomizer()
+    fun weightedRandomizer(): IndexRandomizer = WeightedRandomizer()
```

Comment:
Incidentally fixing this. We should provide the interface, not the concrete implementation